### DISCUSSION TITLE
What does "sparse" mean in least squares context?

### DISCUSSION THREAD

**Discussion by ogencoglu** [qpsolvers/qpsolvers#261] [Q&A] [answered]
Specifically regarding `sparse_conversion` flag. Does it mean `R` is sparse or does it mean we have prior knowledge that the solution we are searching for is sparse?

**Comment by stephane-caron** [comment, marked as answer]
The `sparse_conversion` parameter is a selector for the LS-to-QP conversion strategy, *i.e.* how the QP cost $P, q$ is computed from the LS cost $R, s$. See the parameter details and the notes below in the documentation of [`solve_ls`](https://qpsolvers.github.io/qpsolvers/least-squares.html#qpsolvers.solve_ls)).

There are two meanings of "sparse" that are distinct in this context:

- Whether $R$ is a CSC matrix, in which case we consider it "sparse".
- Whether we use the sparse LS-to-QP conversion strategy.

The `sparse_conversion` parameter refers to the second.
